src/Persistent/Pure.hs

```diff
@@ -425,6 +427,23 @@ findUser = findInById dbUserMap
 getUsers :: Query [User]
 getUsers = view dbUsers
 
+getUsersInClass :: SchoolClass -> Query [User]
+getUsersInClass clss = filter (isClassInRole clss . view userRole) <$> view dbUsers
```

Comment:
Probably a code or abstraction duplication is in the air. Please check and if that is the case, unify the two functions.